specifyweb/frontend/js_src/lib/components/QueryBuilder/Fields.tsx

```diff
@@ -166,15 +166,27 @@ export function QueryFields({
 
   const isBasic = React.useContext(IsQueryBasicContext);
 
-  const fieldSpecsAndFields = queryFieldsToFieldSpecs(baseTableName, fields);
+  const fieldSpecsAndFields = React.useMemo(
+    () => queryFieldsToFieldSpecs(baseTableName, fields),
+    [baseTableName, fields]
+  );
+
   const fieldSpecs = fieldSpecsAndFields.map(
     ([_field, fieldSpec]) => fieldSpec
   );
-  const fieldName = fieldSpecs
-    .map((field) =>
-      generateMappingPathPreview(field.baseTable.name, field.toMappingPath())
-    )
-    .join(' ');
+
+  const fieldName = React.useMemo(
+    () =>
+      fieldSpecs
+        .map((field) =>
+          generateMappingPathPreview(
+            field.baseTable.name,
+            field.toMappingPath()
+          )
+        )
+        .join(' '),
+    [fieldSpecs]
```

Comment:
careful! you need eslint to tell you about bugs like this
fieldSpecs is re-created on every render, so this useMemo is counter-productive.
you should either move fieldSpecs inside this useMemo, or have fieldSpecs call it's own useMemo